src/main/java/net/kaczmarzyk/spring/data/jpa/utils/Converter.java

```diff
@@ -201,13 +215,48 @@ private Boolean convertToBoolean(String value) {
 	}
 	
 	public Date convertToDate(String value) {
+		String dateFormat = getDateFormat(Date.class);
 		try {
 			return new SimpleDateFormat(dateFormat).parse(value);
 		} catch (ParseException e) {
 			throw new ValueRejectedException(value, "invalid date, expected format: " + dateFormat);
 		}
 	}
-
+	
+	public String getDateFormat(Class<?> clazz) {
+		if (dateFormat == null) {
+			return DEFAULT_DATE_FORMATS.get(clazz);
+		}
+		
+		return dateFormat;
+	}
+	
+	public UUID convertToUUID(String value) {
+		try {
+			return UUID.fromString(value);
+		} catch (IllegalArgumentException e) {
+			throw new ValueRejectedException(value, "unparseable uuid");
```

Comment:
add value to the message?

Reply:
or perhaps it's not needed, but in such case it should not be present in the exceptions below: 
```} catch (DateTimeParseException | IllegalArgumentException e) {
			throw new ValueRejectedException(value, "invalid offset date time: " + value + ", expected format: " + dateFormat);```